qiskit/algorithms/quantum_time_evolution/real/implementations/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,388 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from collections import defaultdict
+from itertools import product
+from typing import Union, List, Dict, Optional
+
+from qiskit.algorithms.quantum_time_evolution.real.implementations.trotterization.trotter_ops_validator import (
+    _validate_hamiltonian_form,
+    _validate_input,
+    _is_op_bound,
+)
+from qiskit.algorithms.quantum_time_evolution.real.qrte import Qrte
+from qiskit.circuit import Parameter
+from qiskit.opflow import (
+    OperatorBase,
+    StateFn,
+    Gradient,
+    commutator,
+    SummedOp,
+    PauliSumOp,
+    PauliOp,
+    CircuitOp,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.quantum_info import Pauli
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQrte(Qrte):
+    """ Class for performing Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Y, Zero
+            from qiskit.algorithms.quantum_time_evolution.real.implementations.\
+                trotterization.trotter_qrte import TrotterQrte
+
+            operator = X + Z
+            # LieTrotter with 1 rep
+            trotter_qrte = TrotterQrte()
+            initial_state = Zero
+            time = 1
+            evolved_state = trotter_qrte.evolve(operator, time, initial_state)
+    """
+
+    def __init__(self, product_formula: ProductFormula = LieTrotter()) -> None:
+        """
+        Args:
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                            first order product formula with a single repetition.
+        """
+        self.product_formula = product_formula
+
+    def evolve(
+        self,
+        hamiltonian: Union[Pauli, PauliOp, PauliSumOp],
```

Comment:
Maybe this is covered by your TO-DO list, but in line 129, you use `hamiltonian: Union[SummedOp, PauliOp, OperatorBase]` for typing the hamiltonian input, while here it is `hamiltonian: Union[Pauli, PauliOp, PauliSumOp]`. I understand that this should be coherent.